docs/ast.md

```diff
@@ -306,18 +246,24 @@ Here's the same code after typechecking with helper def macros expanded:
       scala.Some.apply[scala.meta.Tree](Impl.this.internalParent)
     else
       scala.None;
-    def origin: scala.meta.Origin = Impl.this.internalOrigin;
+    def origin: scala.meta.Origin = {
```

Comment:
Makes sense! Off topic: I have checked this with the bug I have with https://github.com/mdemarne/scalameta/blob/f0aeadf75fe49a5f009160b1a06ce9e3620ff60f/tests/src/test/scala/tql/TQLOriginSuite.scala, and it's going through there, so I would think that somehow something is not done properly in copying a tree, i.e. https://github.com/mdemarne/scalameta/blob/master/scalameta/src/main/scala/scala/meta/internal/tql/TraverserHelper.scala#L25 is sometimes true; the tree has changed (even with collect { ... } which copies the tree as is in https://github.com/mdemarne/scalameta/blob/master/scalameta/src/main/scala/scala/meta/tql/Combinators.scala#L174).